src/services/coreBackService.js

```diff
@@ -0,0 +1,83 @@
+const {
+  API_BASE_URL,
+  API_BUILD_PROVEN_USER_IDENTITY_DETAILS,
+  API_CRI_CALLBACK,
+  API_SESSION_INITIALISE
+} = require("../lib/config");
+const https = require("https");
+const axios = require("axios");
+
+class CoreBackService {
+  constructor() {
+    this.axiosInstance = axios.create({
+      httpsAgent: new https.Agent({ keepAlive: true }),
+    })
+  }
+
+  postAction(req, action) {
+    return this.axiosInstance.post(
+      `${API_BASE_URL}/${action}`,
+      {},
+      req.session?.clientOauthSessionId
+        ? this.generateAxiosConfigWithClientSessionId(req)
+        : this.generateAxiosConfig(req),
+    )
+  }
+
+  postSessionInitialise(req, authParams) { return this.axiosInstance.post(
+      `${API_BASE_URL}${API_SESSION_INITIALISE}`,
```

Comment:
Not necessary, but you can actually set a baseURL on the axios instance, which means you only need to specify the path part here